securedrop/source_app/disable.py

```diff
@@ -0,0 +1,19 @@
+from datetime import date
+from flask import render_template, g
+from source_app.decorators import ignore_static
+import i18n
+import platform
+
+
+def disable_app(app):
+
+    @app.before_request
+    @ignore_static
+    def disable_ui():
+        if(platform.linux_distribution()[1] != app.config['XENIAL_VER'] and
+                date.today() > app.config['TRUSTY_DISABLE_DATE']):
+            g.locale = i18n.get_locale(app.sdconfig)
```

Comment:
Are these added here because we haven't yet called the other `before_request` handlers? If this was done to follow my suggestion in the original ticket of running this handler before all other handlers, that was to *prevent* any other code from being hit to minimize code paths on a possibly insecure instance. What we could do is split `setup_g` into multiple smaller handlers that each do one thing like `init_i18n1` and `check_expires` and such. This would make some of the code more reusuable and put less logic into this handler.

Reply:
good point, done